src/VisualStudio/Core/Def/Implementation/LanguageClient/LiveShareLanguageServerClient.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Utilities;
+using Nerdbank.Streams;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
+{
+    [ContentType("C#_LSP")]
+    [ContentType("VB_LSP")]
```

Comment:
?

Reply:
running on the liveshare guest which has content type C#_LSP / VB_LSP.  We don't want to activate this for local experiences.  Liveshare has a work item to allow us to only activate a language server for a guest which will allow us to switch this to c#.

Reply:
Documents opened via Live Share have a separate content type than the regular ones. Yes, it's a hack; yes, it causes problems, and yes we're working to get rid of it.